File: backend/src/Core/Models/CompanyFundamentals/CashFlowStatement.cs
namespace Core.Models.CompanyFundamentals;

public class CashFlowStatement
{
    public required string Symbol { get; set; }
    public required IEnumerable<CashFlowStatementPeriod> PeriodData { get; set; }
}

public class CashFlowStatementPeriod
{
    public required string FiscalYear { get; set; }
    public required string Period { get; set; }
    public long StockBasedCompensation { get; set; }
    public long FreeCashFlow { get; set; }
}

File: apps/web/src/app/shared/models/fundamentals/cash-flow-statement.ts
export type CashFlowStatement = {
  symbol: string;
  periodData: CashFlowStatementPeriod[];
}

export type CashFlowStatementPeriod = {
  fiscalYear: string;
  period: string;
  stockBasedCompensation: number;
  freeCashFlow: number;
}
